project-base/storefront/components/Blocks/Banners/BannersSliderItem.tsx

```diff
@@ -1,35 +1,45 @@
 import { desktopFirstSizes } from 'components/Theme/mediaQueries';
 import { ImageSizeFragmentApi, SliderItemFragmentApi } from 'graphql/generated';
 import { useGetWindowSize } from 'hooks/ui/useGetWindowSize';
+import { forwardRef } from 'react';
+import { Loader } from 'components/Basic/Loader/Loader';
 
 type BannersSliderItemProps = {
     item: SliderItemFragmentApi;
 };
 
-export const BannersSliderItem: FC<BannersSliderItemProps> = ({
-    item: { webMainImage, mobileMainImage, link, name },
-}) => {
-    const { width } = useGetWindowSize();
-    const image = width > desktopFirstSizes.tablet ? webMainImage : mobileMainImage;
+export const BannersSliderItem = forwardRef<HTMLAnchorElement, BannersSliderItemProps>(
+    ({ item: { webMainImage, mobileMainImage, link, name } }, ref) => {
+        const { width } = useGetWindowSize();
+        const image = width > desktopFirstSizes.tablet ? webMainImage : mobileMainImage;
+        const imageSize: ImageSizeFragmentApi | null = image?.sizes.find((i) => i.size === 'default') ?? null;
+        const imageElement = !imageSize ? (
+            <BannerImage src="images/optimized-noimage.webp" alt="no image" />
+        ) : (
+            <picture>
+                {imageSize.additionalSizes.map((additionalSize) => (
+                    <source key={additionalSize.url} srcSet={additionalSize.url} media={additionalSize.media} />
+                ))}
+                <BannerImage src={imageSize.url} alt={image?.name || name} />
+            </picture>
+        );
 
-    const imageSize: ImageSizeFragmentApi | null = image?.sizes.find((i) => i.size === 'default') ?? null;
+        return (
+            <a href={link} className="h-80 min-w-full max-w-full basis-full snap-center" draggable={false} ref={ref}>
+                {width < 0 ? (
+                    <div className="flex h-full w-full items-center justify-center">
+                        <Loader className="w-14" />
+                    </div>
+                ) : (
+                    imageElement
+                )}
```

Comment:
I understand the logic with a spinner and I must say I like the UX of it, as it looks much better than a jumpy slider, but isn't it possible to handle it with just CSS?

Reply:
As I wrote in the previous comment, only possible when we have only one type (size) of the images.